packages/core/admin/admin/src/content-manager/components/DynamicZone/components/Component/index.js

```diff
@@ -56,6 +57,12 @@ const Component = ({
 }) => {
   const { formatMessage } = useIntl();
   const { getComponentLayout } = useContentTypeLayout();
+  const componentLayoutData = useMemo(() => {
+    const layout = getComponentLayout(componentUid);
+
+    return layout;
+  }, [componentUid, getComponentLayout]);
+  const mainValue = useMainValue(componentLayoutData, [name, index]);
```

Comment:
I would recommend to pass the mainValue as a prop from the `DynamicZone/utils/select.js` hook to avoid extreme rerendering of the components, it can cause performance issues at some point.

Reply:
I have created a new wrapper with `utils/connect` for DynamicZone/Component. 

Please check this out. It is okay for using this separately for each Component.
(it feel cleaner for me and the RepeatableComponent/DraggedItem using the same way, if I see it correctly).

Or should I move the mainValue parsing here `DynamicZone/utils/select.js`:

```js
  const dynamicDisplayedComponents = useMemo(
    () => get(modifiedData, [name], []).map(data => data.__component),
    [modifiedData, name]
  );
```